### PR TITLE
WAL-1486: Unit Tests

### PR DESCRIPTION
**Pull request by lluiserena**
## Description

Added unit tests on Crossmint-SDK AA.

Captura de pantalla 2023-12-20 a las 15 44 26

 Coverage:  47.72% lines covered

## Test plan

yarn run test
yarn run test-coverage

### CODE REVIEW

File: packages/client/wallets/aa/src/api/CrossmintService.test.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+    describe("fetchCrossmintAPI", () => {
+        it("handles successful API call", async () => {
+            // Test successful API call handling
+        });
+
+        it("handles API errors correctly", async () => {
+            // Test API error handling
+        });
+    });
```
Comment (begona-paella) [inline packages/client/wallets/aa/src/api/CrossmintService.test.ts:176]: when do you plan to add these tests?

**Review by begona-paella — COMMENTED** [review]
Can you specify in the description the % of coverage?

File: packages/client/wallets/aa/src/api/CrossmintService.test.ts (hunk 1 above)
Comment (lluiserena) [reply packages/client/wallets/aa/src/api/CrossmintService.test.ts:176]: Pushing it in a few minutes. Forgot to update it

Comment (lluiserena) [reply packages/client/wallets/aa/src/api/CrossmintService.test.ts:416]: Pushed

**Comment by jordiamez** [comment]
Updated PR

File: packages/client/wallets/aa/src/CrossmintAASDK.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    describe("init", () => {
+        it("should initialize the SDK correctly", () => {
+            expect(sdk).toBeInstanceOf(CrossmintAASDK);
+        });
+    });
+
+    describe("getOrCreateWallet", () => {
+        const mockUser: UserIdentifier = { email: "test@example.com" };
+        const mockChain = "ethereum"; // Replace with a valid blockchain value
+        const mockWalletConfig: WalletConfig = {
```
[the remark this answers is not in the document]
Comment (jordiamez) [reply packages/client/wallets/aa/src/CrossmintAASDK.test.ts:33]: Doesn't make change to test with other changes, do you think we must include all chains?

File: packages/client/wallets/aa/src/CrossmintAASDK.test.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            signer: {
+                type: "FIREBLOCKS_NCW",
```
[the remark this answers is not in the document]
Comment (jordiamez) [reply packages/client/wallets/aa/src/CrossmintAASDK.test.ts:35]: Yes does it make a lot of sense

File: packages/client/wallets/aa/src/utils/auth.test.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            name: "John Doe",
+            iat: 1516239022,
+        };
+
+        const result = parseToken(sampleToken);
+        expect(result).toEqual(expectedPayload);
+    });
+
+    it.skip("throws an error for an invalid token", () => {
+        const invalidToken = "invalid.token";
```
[the remark this answers is not in the document]
Comment (jordiamez) [reply packages/client/wallets/aa/src/utils/auth.test.ts:36]: Removed skip()

File: packages/client/wallets/aa/src/utils/auth.test.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        });
+    });
+
+    afterAll(() => {
+        // Restaura window a su estado original
+        Object.assign(window, originalWindow);
+    });
+
+    it.skip("correctly parses a valid token", () => {
+        const sampleToken =
```
[the remark this answers is not in the document]
Comment (jordiamez) [reply packages/client/wallets/aa/src/utils/auth.test.ts:23]: Removed skip()

File: packages/client/wallets/aa/src/utils/error.test.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            expect(error.code).toBe("ERROR_NOT_AUTHORIZED");
+            expect(error).toBeInstanceOf(NotAuthorizedError);
+            expect(error).toBeInstanceOf(Error);
+        });
+    });
+
+    // Test for RateLimitError
+    describe("RateLimitError", () => {
+        it("should create an instance with the correct message and code", () => {
+            const error = new RateLimitError("Too many requests", 1000);
```
[the remark this answers is not in the document]
Comment (jordiamez) [reply packages/client/wallets/aa/src/utils/error.test.ts:33]: Done

File: packages/client/wallets/aa/src/utils/auth.test.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (jordiamez) [reply packages/client/wallets/aa/src/utils/auth.test.ts:32]: Removed skip and fixed test

**Comment by jordiamez** [comment]
Captura de pantalla 2024-01-08 a las 12 25 10

**Review by begona-paella — APPROVED** [review]

**Review by jordiamez — APPROVED** [review]

**Pull request merged**